### PR TITLE
bugfix(core): Improve data flow, clean code, integrate autosave with …

### PR DESCRIPTION
**Pull request by derpdead**
…user avatar

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Issue # (issue)

## Type of change

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [ ] Unit Tests
- [ ] e2e Test

# Checklist:

- [x] I have read the contribution requirements and fulfil them.
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes

### CODE REVIEW

File: modules/@ergonode/ui/src/components/StaticImage/StaticImage.vue [hunk 1]
```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright © Bold Brand Commerce Sp. z o.o. All rights reserved.
+ * See LICENSE for license details.
+ */
+<template>
+    <img
+        ref="img"
+        :style="styles"
+        :class="classes">
```
Comment (tomulus91) [inline modules/@ergonode/ui/src/components/StaticImage/StaticImage.vue:9]: nie powinno być '/' na koncu ?

Comment (derpdead) [reply modules/@ergonode/ui/src/components/StaticImage/StaticImage.vue:9]: img is called as Void element, doesnt require to close it.

File: modules/@ergonode/users/src/components/Tabs/UserAvatarTab.vue [hunk 2]
```diff
@@ -5,20 +5,91 @@
...
+    methods: {
+        ...mapActions('authentication', [
+            'getUser',
+        ]),
+        ...mapActions('user', [
+            'updateUserAvatar',
+        ]),
+        onSubmit() {
+            if (this.isSubmitting) {
+                return;
```
Comment (tomulus91) [inline modules/@ergonode/users/src/components/Tabs/UserAvatarTab.vue:62]: taki dziwny ten return.... nie lepiej wywalic return i zrobic !this.isSubmitting { kod }

Comment (derpdead) [reply modules/@ergonode/users/src/components/Tabs/UserAvatarTab.vue:62]: It's how we do our code styling

**Review by tomulus91 — APPROVED** [review]

**Review by wiewiurdp — APPROVED** [review]

**Pull request merged**